### PR TITLE
Threading logic: merge sbarratt/threading

### PR DESCRIPTION
**Pull request by sbarratt**
Since SCS releases the GIL, we can simply wrap the diffcp function with multi-threading .

### CODE REVIEW

File: diffcp/cone_program.py [hunk 1]
```diff
@@ -28,6 +30,11 @@ def dpi(z, cones):
         sparse.diags(.5 * (np.sign(w) + 1))
     ])
 
+def solve_and_derivative_batch(As, bs, cs, cone_dicts, n_jobs=4, warm_starts=None, **kwargs):
+    batch_size = len(As)
+    return Parallel(n_jobs=n_jobs, backend="threading")(
+        delayed(solve_and_derivative)(As[i], bs[i], cs[i], cone_dicts[i], warm_start=None if warm_starts is None else warm_starts[i], **kwargs) for i in (range(batch_size)))
```
Comment (akshayka) [inline diffcp/cone_program.py]: nit, lines shouldn't exceed 80 columns

File: tests.py [hunk 2]
```diff
@@ -279,6 +280,36 @@ def test_warm_start(self):
...
+            results.append(cone_prog.solve_and_derivative(A, b, c, cone_dims))
+            toc = time.time()
+            serial_time += toc - tic
+
+        tic = time.time()
+        results_thread = cone_prog.solve_and_derivative_batch(As, bs, cs, cone_dicts)
+        toc = time.time()
+        parallel_time = toc - tic
+
+        print(parallel_time, serial_time)
```
Comment (akshayka) [inline tests.py]: nit, if you want to print these stats, include some text (e.g., "parallel time {:f}, serial time {:f}".format(parallel_time, serial_time))

**Comment by sbarratt** [comment]
Ok, I've incorporated these changes. I've also added the newest version of scs to the setup.py requirements. This stops the AA warning messages being printed. (https://github.com/cvxgrp/scs/issues/115)

**Comment by akshayka** [comment]
Thanks!

**Pull request merged**